src/Chainweb/Version.hs

```diff
@@ -624,17 +624,42 @@ vuln797FixDate Testnet04 = epoch
 vuln797FixDate Mainnet01 = [timeMicrosQQ| 2019-12-10T21:00:00.0 |]
 {-# INLINE vuln797FixDate #-}
 
--- | Upgrade coin v2 at time, or at block height 1
--- | Must be BEFORE 'txEnabledDate' in mainnet.
-upgradeCoinV2Date :: ChainwebVersion -> Maybe (Time Micros)
-upgradeCoinV2Date Test{} = Nothing
-upgradeCoinV2Date TimedConsensus{} = Nothing
-upgradeCoinV2Date PowConsensus{} = Nothing
-upgradeCoinV2Date TimedCPM{} = Nothing
-upgradeCoinV2Date FastTimedCPM{} = Nothing
-upgradeCoinV2Date Development = Just [timeMicrosQQ| 2019-12-14T18:50:00.0 |]
-upgradeCoinV2Date Testnet04 = Nothing
-upgradeCoinV2Date Mainnet01 = Just [timeMicrosQQ| 2019-12-17T15:00:00.0 |]
+-- | Mainnet upgrade to coin v2 at time at @[timeMicrosQQ| 2019-12-17T15:00:00.0 |]@,
+-- which was BEFORE 'txEnableDate'.
+--
+-- This function provides the block heights when coin v2 became effective on the
+-- respective chains.
+--
+coinV2Upgrade
```

Comment:
Is this really better? We're replacing a single date with a bunch of constants.

Reply:
I think, the logic at the use site becomes simpler.

Reply:
How did you find the numbers? This is not reviewable code

Reply:
We can keep this in the interests of expediency, but I don't see much value in this change so I question why you bothered to do the necessary work. I'm also concerned that there is no way to validate these heights except "run it in mainnet".